include/wil/common.h

```diff
@@ -372,19 +372,19 @@ check fails as opposed to the invalid parameter handler that the STL invokes. Th
 #endif
 
 /// @cond
-#if (__cplusplus >= 201703) || (_MSVC_LANG >= 201703)
-#define WIL_HAS_CXX_17 1

```

Comment:
Should the cpp17 check have an `#error` if false?  It would be nice for people who make that mistake to get a really clear/obvious error instead of a ton of confusing spew.

Reply:
I removed this instance to reduce the number of macros/different checks we have for C++ standard version to just use the `__WI_LIBCPP_STD_VER` macro.

Or are you suggesting that we introduce an error when not building as C++17 (i.e. enforce a hard minimum). Maybe... but I'm not sure how much that might break. For the longest time C++14 was the minimum. It wasn't until somewhat recently that we bumped the minimum to C++17 for the tests (I.e. we're no longer ensuring C++14 builds, effectively bumping up the "soft" minimum), however I'm not quite sure that we can do that without breaking some dependent projects. Perhaps we should, and just tell them to update their version if they wish to update their version of WIL, however we might want to take a slightly more careful approach before we do that.

Reply:
Ah, so cpp17 is the minimum _for the tests_ but not necessarily for all consumers of WIL.

If cpp14 is the hard floor then maybe detect and error in that case?  Or is the hard floor even lower than that?